code/modules/research/xenobiology/crossbreeding/reproductive.dm

```diff
@@ -3,47 +3,31 @@ Reproductive extracts:
 	When fed three monkey cubes, produces between
 	1 and 4 normal slime extracts of the same colour.
 */
+
+
 /obj/item/slimecross/reproductive
 	name = "reproductive extract"
 	desc = "It pulses with a strange hunger."
 	icon_state = "reproductive"
 	effect = "reproductive"
 	effect_desc = "When fed monkey cubes it produces more extracts. Bio bag compatible as well."
 	var/extract_type = /obj/item/slime_extract/
-	var/cubes_eaten = 0
+	var/cooldown = 3 SECONDS
 	var/last_produce = 0
-	var/cooldown = 30 // 3 seconds.
+
+/obj/item/slimecross/reproductive/Initialize()
+	. = ..()
+	LoadComponent(/datum/component/storage/concrete/extract_inventory)
 
 /obj/item/slimecross/reproductive/attackby(obj/item/O, mob/user)
 	if((last_produce + cooldown) > world.time)
 		to_chat(user, span_warning("[src] is still digesting!"))
 		return
 	if(istype(O, /obj/item/storage/bag/bio))
 		var/list/inserted = list()
-		SEND_SIGNAL(O, COMSIG_TRY_STORAGE_TAKE_TYPE, /obj/item/food/monkeycube, src, 1, null, null, user, inserted)
+		SEND_SIGNAL(O, COMSIG_TRY_STORAGE_TAKE_TYPE, /obj/item/food/monkeycube, src, 3, null, null, user, inserted)
 		if(inserted.len)
-			var/obj/item/food/monkeycube/M = inserted[1]
-			if(istype(M))
-				eat_cube(M, user)
-		else
-			to_chat(user, span_warning("There are no monkey cubes in the bio bag!"))
-	if(istype(O,/obj/item/food/monkeycube))
-		eat_cube(O, user)
-	if(cubes_eaten >= 3)
-		var/cores = rand(1,4)
-		visible_message(span_notice("[src] briefly swells to a massive size, and expels [cores] extract[cores > 1 ? "s":""]!"))
-		playsound(src, 'sound/effects/splat.ogg', 40, TRUE)
-		last_produce = world.time
-		for(var/i = 0, i < cores, i++)
-			new extract_type(get_turf(loc))
-		cubes_eaten = 0
-
-/obj/item/slimecross/reproductive/proc/eat_cube(obj/item/food/monkeycube, mob/user)
-		qdel(monkeycube)
-		cubes_eaten++
-		to_chat(user, span_notice("You feed [monkeycube] to [src], and it pulses gently."))
-		playsound(src, 'sound/items/eatfood.ogg', 20, TRUE)
-
+			SEND_SIGNAL(src, COMSIG_TRY_STORAGE_CONSUME_CONTENTS)
```

Comment:
```suggestion
			SEND_SIGNAL(src, COMSIG_TRY_STORAGE_CONSUME_CONTENTS)
			
```
this one isnt a big deal, but its best to have a space between procs/type paths, so they dont look like one big block, its easier to read and navigate when you can visually seperate them